plugins/signals-backend/src/service/SignalManager.ts

```diff
@@ -64,6 +68,46 @@ export class SignalManager {
       onEvent: (params: EventParams) =>
         this.onEventBrokerEvent(params.eventPayload as SignalPayload),
     });
+
+    if (
+      options.config.getOptionalBoolean('signals.server.pingEnabled') ??
+      true
+    ) {
+      const pingIntervalConfig = options.config.getOptionalConfig(
+        'signals.server.pingInterval',
+      );
+      this.pingInterval = setInterval(
+        () => this.ping(),
+        pingIntervalConfig
+          ? durationToMilliseconds(readDurationFromConfig(pingIntervalConfig))
+          : 30000,
+      );
+    }
+  }
+
+  private ping() {
+    this.connections.forEach(conn => {
+      if (!conn.isAlive) {
+        this.logger.info(`Connection ${conn.id} is not alive, closing`);
+        conn.ws.terminate();
+        return;
+      }
+
+      conn.isAlive = false;
```

Comment:
instead of `isAlive` as boolean we could store a timestamp when the last message was received/sent, so that we can skip pinging in case an event has been sent recently

Reply:
But sending an event doesn't mean the connection is alive. Instead ping with pong response does that. And these are different frames

Reply:
If we receive any messages we could do the same at least, but I think we can leave that for later if we feel it would improve something. The current global interval is simple and adding in this logic would complicate things quite a bit